README.md

```diff
@@ -1,21 +1,50 @@
 # Workflow Status Slack Notification Action
 
-> Github action that keeps track and retrieves previous workflow runs status for the same workflow on the same branch. 
-  According to current and previous runs status, it will send a notification to a Slack channel using an Incoming webhook app.
+> Github action that keeps track and retrieves previous workflow runs status for the same workflow on the same branch.
+> According to current and previous runs status, it will send a notification to a Slack channel using an Incoming webhook app.
 
 ## How it works
 
 At the end of a workflow, on success() and on failure(), call this action with the current status as a parameter (success/failure).
-This action will first check the previous status and according to the current status. It will send a notification to Slack if:
+This action will first check the previous workflow status and will send a Slack notification according to the current workflow status.
+Status is retrieved from cache if the current workflow has already run previously (from `re-run all jobs` button for example).
+If it's the first time this workflow runs, this action will retrieve previous workflow status on the same branch with `gh` cli using provided `github-token`.
+
+It'll compare last/current workflow status and will send a notification to Slack if:
+
 - Current status is `failure` and it's the first time this workflow runs for the current branch.
 - Current status is `failure` and previous run status was `success`.
 - Current status is `success` and previous run status was `failure`.
```

Comment:
Do you see any value in send a notification if it's the first run and it's a success?

Reply:
No that's the thing, we only send a success notification if the current workflow is a success and the previous one was a failure.
If the current workflow is a success and it's the first one or the previous one was a success, we don't send a notification.
I can clarify this by adding in the README what doesn't trigger a notification

Reply:
That makes sense. I didn't know if it was desired to send a slack notif when the first was successful. Keeping the noise down is good because these notifications can flood slack, so I like the approach we're taking to only notify when things change.